### PR TITLE
Fix issue 12341

### LINKED ISSUE #12341: [REQ] isVoid flag for Java templates
Hello dear colleagues! Can you help, please with my question: at now is it possible to check the type for "isVoid" in Java templates? I generate Spring controllers with standard openapi-generator [Java-Spring](https://github.com/OpenAPITools/openapi-generator/tree/master/modules/openapi-generator/src/main/resources/JavaSpring) templates and I get the Void type as a result. For example:
```
Void delete(
        @ApiParam(value = "User View Section",required=true) @PathVariable("section") String section
);
```
I want to change Void type for primitive void. In [api.mustache](https://github.com/OpenAPITools/openapi-generator/blob/master/modules/openapi-generator/src/main/resources/JavaSpring/api.mustache) template I can't override it. Also I try to edit the [returnTypes.mustache](https://github.com/OpenAPITools/openapi-generator/blob/master/modules/openapi-generator/src/main/resources/JavaSpring/returnTypes.mustache) template and it doesn't help.

I try to update returnTypes.mustache template like:
```
{{#isMap}}Map<String, {{{returnType}}}>{{/isMap}}{{#isArray}}{{^reactive}}{{{returnContainer}}}{{/reactive}}<{{{returnType}}}>{{/isArray}}{{^returnContainer}}{{#isVoid}}void{{/isVoid}}{{^isVoid}}{{{returnType}}}{{/isVoid}}{{/returnContainer}}
```

How I can get void instead of Void?

### PR DESCRIPTION
**Pull request by rodrigoma3**
if method return type is void then no return.
Unit tests to verify that the API methods were created as expected.
fix #12341 
First commit has the changes made to fix the issue

### PR checklist
 
- [ X ] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [ X ] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [ X ] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [ ] In case you are adding a new generator, run the following additional script : 
  ```
  ./bin/utils/ensure-up-to-date.sh
  ``` 
  Commit all changed files.
- [ X ] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (6.3.0) (minor release - breaking changes with fallbacks), `7.0.x` (breaking changes without fallbacks)
- [ X ] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

@cachescrubber (2022/02) @welshm (2022/02) @MelleD (2022/02) @atextor (2022/02) @manedev79 (2022/02) @javisst (2022/02) @borsch (2022/02) @banlevente (2022/02) @Zomzog (2022/09)

### CODE REVIEW

File: modules/openapi-generator/src/test/java/org/openapitools/codegen/java/assertions/MethodAssert.java [hunk 1]
```diff
@@ -71,6 +71,23 @@ public MethodAssert bodyContainsLines(final String... lines) {
...
+    public MethodAssert bodyNotContainsLines(final String... lines) {
+        Assertions.assertThat(isWithImplementation())
+            .withFailMessage("Method %s is abstract", methodSignature)
+            .isTrue();
+        final String actualBody = actual.getTokenRange()
+            .orElseThrow(() -> new IllegalStateException("Not-abstract method doesn't have body"))
+            .toString();
+        Assertions.assertThat(actualBody)
+            .withFailMessage(
+                "Method's %s body should contains lines\n====\n%s\n====\nbut actually was\n====\n%s\n====",
```
Comment (borsch) [inline modules/openapi-generator/src/test/java/org/openapitools/codegen/java/assertions/MethodAssert.java:83]: `should` -> `shouldn't`

File: docs/generators/fsharp-functions.md [hunk 2]
```diff
@@ -34,7 +34,7 @@ These options may be applied as additional-properties (cli) or configOptions (pl
 |prependFormOrBodyParameters|Add form or body parameters to the beginning of the parameter list.| |false|
 |sortModelPropertiesByRequiredFlag|Sort model properties to place required parameters before optional parameters.| |true|
 |sortParamsByRequiredFlag|Sort method arguments to place required parameters before optional parameters.| |true|
-|sourceFolder|source folder for generated code| |OpenAPI/src|
+|sourceFolder|source folder for generated code| |OpenAPI\src|
```
Comment (borsch) [inline docs/generators/fsharp-functions.md:37]: re-generate all samples under unix/docker/WSL to fix these changes

File: modules/openapi-generator-core/pom.xml [hunk 3]
```diff
@@ -6,7 +6,7 @@
     <artifactId>openapi-generator-project</artifactId>
     <groupId>org.openapitools</groupId>
     <!-- RELEASE_VERSION -->
-    <version>6.6.0-SNAPSHOT</version>
+    <version>6.6.0</version>
```
Comment (borsch) [inline modules/openapi-generator-core/pom.xml:9]: pull latest master

**Review by borsch — CHANGES_REQUESTED** [review]

**Comment by rodrigoma3** [comment]
@borsch resolved, could you look again and signal me if it's ok now please?

File: samples/client/echo_api/python-nextgen/README.md [hunk 4]
```diff
@@ -85,35 +85,35 @@ All URIs are relative to *http://localhost:3000*
...
-*QueryApi* | [**test_query_style_deep_object_explode_true_object_all_of**](docs/QueryApi.md#test_query_style_deep_object_explode_true_object_all_of) | **GET** /query/style_deepObject/explode_true/object/allOf | Test query parameter(s)
-*QueryApi* | [**test_query_style_form_explode_true_array_string**](docs/QueryApi.md#test_query_style_form_explode_true_array_string) | **GET** /query/style_form/explode_true/array_string | Test query parameter(s)
-*QueryApi* | [**test_query_style_form_explode_true_object**](docs/QueryApi.md#test_query_style_form_explode_true_object) | **GET** /query/style_form/explode_true/object | Test query parameter(s)
-*QueryApi* | [**test_query_style_form_explode_true_object_all_of**](docs/QueryApi.md#test_query_style_form_explode_true_object_all_of) | **GET** /query/style_form/explode_true/object/allOf | Test query parameter(s)
+*BodyApi* | [**test_binary_gif**](docs\BodyApi.md#test_binary_gif) | **POST** /binary/gif | Test binary (gif) response body
```
Comment (borsch) [inline samples/client/echo_api/python-nextgen/README.md:88]: doesn't seems like you have fixed this

File: samples/openapi3/server/petstore/springboot-delegate/src/main/java/org/openapitools/api/FakeApi.java [hunk 5]
```diff
@@ -72,7 +72,7 @@ default FakeApiDelegate getDelegate() {
     default ResponseEntity<Void> createXmlItem(
         @Parameter(name = "XmlItem", description = "XmlItem Body", required = true) @Valid @RequestBody XmlItem xmlItem
     ) {
-        return getDelegate().createXmlItem(xmlItem);
+            getDelegate().createXmlItem(xmlItem);
```
Comment (borsch) [inline samples/openapi3/server/petstore/springboot-delegate/src/main/java/org/openapitools/api/FakeApi.java:75]: doesn't seems to be correct

File: modules/openapi-generator/src/main/resources/JavaSpring/api.mustache [hunk 6]
```diff
@@ -237,7 +237,7 @@ public interface {{classname}} {
         {{/hasParams}}{{#springFoxDocumentationProvider}}@ApiIgnore {{/springFoxDocumentationProvider}}{{#springDocDocumentationProvider}}@ParameterObject {{/springDocDocumentationProvider}}final Pageable pageable{{/vendorExtensions.x-spring-paginated}}
     ){{#unhandledException}} throws Exception{{/unhandledException}}{{^jdk8-default-interface}};{{/jdk8-default-interface}}{{#jdk8-default-interface}} {
         {{#delegate-method}}
-        return {{operationId}}({{#allParams}}{{paramName}}{{^-last}}, {{/-last}}{{/allParams}}{{#reactive}}{{#hasParams}}, {{/hasParams}}exchange{{/reactive}}{{#vendorExtensions.x-spring-paginated}}, pageable{{/vendorExtensions.x-spring-paginated}});
+            {{^isVoid}}return {{/isVoid}}{{#isVoid}}{{#useResponseEntity}}return {{/useResponseEntity}}{{^useResponseEntity}}{{#reactive}}return {{/reactive}}{{/useResponseEntity}}{{/isVoid}}{{operationId}}({{#allParams}}{{paramName}}{{^-last}}, {{/-last}}{{/allParams}}{{#reactive}}{{#hasParams}}, {{/hasParams}}exchange{{/reactive}}{{#vendorExtensions.x-spring-paginated}}, pageable{{/vendorExtensions.x-spring-paginated}});
```
Comment (borsch) [inline modules/openapi-generator/src/main/resources/JavaSpring/api.mustache:240]: remove spaces before `{{^isVoid}}`

File: modules/openapi-generator/src/main/resources/JavaSpring/api.mustache [hunk 7]
```diff
@@ -247,7 +247,7 @@ public interface {{classname}} {
         {{>methodBody}}
         {{/isDelegate}}
         {{#isDelegate}}
-        return getDelegate().{{operationId}}({{#allParams}}{{paramName}}{{^-last}}, {{/-last}}{{/allParams}}{{#reactive}}{{#hasParams}}, {{/hasParams}}exchange{{/reactive}}{{#vendorExtensions.x-spring-paginated}}, pageable{{/vendorExtensions.x-spring-paginated}});
+            {{^isVoid}}return {{/isVoid}}getDelegate().{{operationId}}({{#allParams}}{{paramName}}{{^-last}}, {{/-last}}{{/allParams}}{{#reactive}}{{#hasParams}}, {{/hasParams}}exchange{{/reactive}}{{#vendorExtensions.x-spring-paginated}}, pageable{{/vendorExtensions.x-spring-paginated}});
```
Comment (borsch) [inline modules/openapi-generator/src/main/resources/JavaSpring/api.mustache:250]: remove spaces before `{{^isVoid}}`

**Review by borsch — CHANGES_REQUESTED** [review]

**Comment by rodrigoma3** [comment]
@borsch Sorry, something happened here and it messed up a lot. I fixed it and it looks right now. Thanks

**Review by borsch — APPROVED** [review]

**Pull request merged**